File: Top_Interview_150/Array_String/cpp/274.cpp
#include <vector>
#include <algorithm>

using namespace std;

class Solution
{
public:
    int hIndex(vector<int> &citations)
    {
        sort(citations.begin(), citations.end());
        int n = citations.size();

        int j = 0;
        int h = -1;
        for (int i = 0; i < 1000; ++i)
        {
            while (j < n && citations[j] < i)
            {
                j++;
            }

            if (n - j >= i)
            {
                h = i;
            }
        }

        return h;
    }
};

File: Top_Interview_150/Array_String/python/274.py
from typing import List


class Solution:
    def hIndex(self, citations: List[int]) -> int:
        citations.sort()
        n = len(citations)

        j = 0
        h = -1
        for i in range(1000):
            while j < n and citations[j] < i:
                j += 1

            if n - j >= i:
                h = i

        return h


solution = Solution()
print(solution.hIndex(citations = [3,0,6,1,5]))
